pkg/virtual/framework/rootapiserver/options/options.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"github.com/spf13/pflag"
+
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+
+	kcpclient "github.com/kcp-dev/kcp/pkg/client/clientset/versioned"
+	kcpinformer "github.com/kcp-dev/kcp/pkg/client/informers/externalversions"
+	"github.com/kcp-dev/kcp/pkg/virtual/framework"
+	"github.com/kcp-dev/kcp/pkg/virtual/framework/rootapiserver"
+	workspacesoptions "github.com/kcp-dev/kcp/pkg/virtual/workspaces/options"
+)
+
+const virtualWorkspacesFlagPrefix = "virtual-workspaces-"
+
+type Root struct {
+	Workspaces *workspacesoptions.Workspaces
```

Comment:
am curious, why do we need the root options in the first place? Why not leave it as it is and just put a Syncer options struct next to Workspaces?

Note: having these in framework seems wrong.





[13:26](https://coreos.slack.com/archives/DNS57BDAL/p1652268363286879)
Why not just a Syncer beside Workspaces?